scripts/commits-check.sh

```diff
@@ -0,0 +1,31 @@
+#!/usr/bin/env bash
+#
+# This script retrieves the list of commits difference from master
+# and ensures that each commit starts with a MGDAPI- numbered ticket
+# See https://issues.redhat.com/browse/MGDAPI-3097
+
+merge_base=$(git merge-base master HEAD)
+commits=$(git rev-list --no-merges $merge_base..HEAD)
+invalidCommits=()
+
+if [ -z "$commits" ]; then
+    echo "No commits to scan"
+    exit 0
+fi
+
+for sha in $commits; do
+    msg=$(git log --format=%B -n 1 $sha)
+    if [[ ! $msg =~ ^(MGDAPI-[0-9]+)[[:blank:]].*$ ]]; then
+        invalidCommits+=( "Commit $sha:\n$msg\n" )
+    fi
+done
+
+if [ ${#invalidCommits[@]} -gt 0 ]; then
+    printf "Not all commits start wtih a \"MGDAPI-\" ticket\n\n"
+    for i in "${invalidCommits[@]}"; do
+        printf "$i"
+    done
+    exit 1
```

Comment:
Does it make sense to exit with an error code of `1` because the script technically hasn't failed at this point? Or is the `exit 1` meant to highlight to the user that there is an error with their commits?

Reply:
I made the assumption that the script has to exit with a non-zero exit code for the prow job to detect that this has not passed. I just chose 1 as it's the general error type. Open to suggestions if you think there's a neat alternative though

Reply:
Oh I didn't realize that was for prow to detect it wasn't passing. I think 1 makes sense for the error type so no action is needed on this.